src/mono/wasm/runtime/workers/dotnet-crypto-worker.ts

```diff
@@ -134,28 +140,24 @@ class ChannelWorker {
                 // Reset the size and transition to await state.
                 Atomics.store(this.comm, this.MSG_SIZE_IDX, 0);
                 this._change_state_locked(this.STATE_AWAIT);
-            } finally {
-                this._release_lock();
-            }
+            });
+            if (done) break;
```

Comment:
```suggestion
            if (done)
                break;
```

It's easier to read if the body of an if is on a separate line.